### ISSUE TITLE
Changing date outside of component doesnt affect the component

### ISSUE THREAD

**Issue by padsbanger** [Adphorus/react-date-range#112]
Hello, I have a scenario where date is a React state, that can be changed anywhere in the app. However when it happens the date on component stays the same (it wont update).

Here is the minimalistic scenario from my app. A button that when clicked increases the date by one month.

```js

import React, { Component } from 'react';
import moment from 'moment';
import { Calendar } from 'react-date-range';

export class CalendarView extends Component {
  constructor(props) {
    super(props);
    this.state = {
      dateNow: moment(),
    };
  }

  render() {
    const { dateNow } = this.state;
    return (
      <div className="dashboard-calendar">
      <button
        onClick={() => {
          this.setState({
            dateNow: this.state.dateNow.add(1, 'months'),
          });
        }}
      >Change date
      </button>
      <RangeCalendar
        date={this.state.dateNow}
        format={'Y-MM-DD'}
      />
      </div>
    );
  }
}



```

**Comment by burakcan** [comment]
This should be fixed on 0.9.3

**Comment by kutyel** [comment]
This issue should be reopened as proved in #158 

I am also suffering from the same erratic behaviour and it is quite frustrating...
